apis/duck/v1/kresource_type.go

```diff
@@ -88,3 +90,13 @@ func (t *KResource) GetTypeMeta() *metav1.TypeMeta {
 func (t *KResource) GetStatus() *Status {
 	return &t.Status
 }
+
+// GetHappyConditionType retrieves the happy condition of this resource. Implements the KRShaped interface.
+func (t *KResource) GetHappyConditionType() apis.ConditionType {
+	// Note: KResources are unmarshalled from existing resources. This will only work properly for resources that
+	// have already been initialized to their type.
+	if cond := t.Status.GetCondition(apis.ConditionSucceeded); cond != nil {
```

Comment:
this will not work for un-initialized resources, the condition set will be an empty list.

Reply:
sorry I did not see the final impl, this will work but it wil also require all downstreams to update

Reply:
I have that included as a comment. The idea is that each API shape reports its type and gets initialized. KResource is a generic struct we unmarshal to, but not something that directly manages init.